src/main/java/org/gaul/s3proxy/S3ProxyHandler.java

```diff
@@ -1386,6 +1389,10 @@ private void handlePutBlob(HttpServletRequest request,
                 }
             }
 
+            if (blobStoreType.equals("azureblob")) {
+                eTag = BaseEncoding.base16().lowerCase()
+                        .encode(BaseEncoding.base64().decode(contentMD5String));
+            }
```

Comment:
This returns the (possibly-null) MD5 client header back to the client.  Instead we should return the MD5 header from the server, allowing the client to do end-to-end verification of uploaded data.